x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_custom_dashboards_permissions.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { useKibanaContextForPlugin } from '../../../hooks/use_kibana';
+
+export const useCanManipulateCustomDashboard = () => {
+  const {
+    services: {
+      application: { capabilities },
+    },
+  } = useKibanaContextForPlugin();
+
+  const canLinkOrEditCustomDashboard = useMemo(
+    () => capabilities.savedObjectsManagement.edit === true,
+    [capabilities]
+  );
+
+  const canDeleteCustomDashboard = useMemo(
+    () => capabilities.savedObjectsManagement.delete === true,
+    [capabilities]
+  );
+
+  return {
+    canLinkOrEditCustomDashboard,
+    canDeleteCustomDashboard,
```

Comment:
nit
```suggestion
    canSave,
    canDelete,
```

Reply:
Since the hooks returns the SO capabilities WDYT of making the hook more generic and not linked to dashboard?.

Reply:
Good idea, thank you I renamed the hook to `useSavedObjectsPermissions` as it makes more sense in this case 👍